napari/utils/io.py

```diff
@@ -290,13 +291,59 @@ def write_csv(
             writer.writerow(row)
 
 
+def csv_to_layer_data(filename: str) -> List[FullLayerData]:
+    """Return layer data from a csv.
+
+    Parameters
+    ----------
+    filename : str
+        Filename for reading csv.
+
+    Returns
+    -------
+    list of napari.types.FullLayerData or None
+        Layer full layer data or None.
+    """
+    # If extension is not csv then cannot read file
+    if not filename.endswith('.csv'):
+        return None
+
+    gen = iter_csv(filename)
+    column_names = next(gen)
+    layer_type = guess_layer_type_from_column_names(column_names)
+    if layer_type:
+        print('asdf')
+        print(csv_reader_functions[layer_type])
+        return [csv_reader_functions[layer_type](filename)]
+    else:
+        return None
+
+
+def iter_csv(filename: str):
+    """Iteratively read lines from a csv file.
+
+    Parameters
+    ----------
+    filename : str
+        Filename for reading csv.
+
+    Returns
+    -------
+    generator
+        Generator for reading lines from a csv
+    """
+    with open(filename, newline='') as csvfile:
+        reader = csv.reader(csvfile, delimiter=',')
+        yield from reader
+
+
 def read_csv(filename: str) -> Tuple[np.array, List[str]]:
```

Comment:
could consider removing `read_csv` altogether, I think the only place it is currently used is in `test_read_csv`

Reply:
Agreed - it might be useful though, I'm inclined to keep it for now